cmd/gazelle/fix-update.go

```diff
@@ -322,11 +323,22 @@ func runFixUpdate(wd string, cmd command, args []string) (err error) {
 			mappedKinds    []config.MappedKind
 			mappedKindInfo = make(map[string]rule.KindInfo)
 		)
-		for _, r := range gen {
-			if repl, ok := c.KindMap[r.Kind()]; ok {
+		// We apply map_kind to all rules, including pre-existing ones.
+		var allRules []*rule.Rule
+		allRules = append(allRules, gen...)
```

Comment:
```suggestion
		allRules := gen
```

Reply:
This actually is semantically different code, and can potentially lead to overwriting values if both `allRules` and `gen` end up getting mutated because of the shared buffer. e.g. in this program:

```go
package main

import "fmt"

func main() {
	slice := make([]string, 0, 2)
	slice = append(slice, "original")

	other := slice

	slice = append(slice, "written-to-slice")
	fmt.Printf("slice: %v, other: %v\n", slice, other)
	other = append(other, "written-to-other")
	fmt.Printf("slice: %v, other: %v\n", slice, other)
}
```

the append to `other` ends up overwriting a value in `slice` as well, which we don't want to do here:
```
slice: [original written-to-slice], other: [original]
slice: [original written-to-other], other: [original written-to-other]
```

(While we don't append to `gen` right now, I'd like to guard against this kind of bug in the future, particularly as you only actually hit issues at capacity thresholds, so it's easy to miss in testing)